src/nvim/shada.c

```diff
@@ -3482,6 +3489,8 @@ static inline char *get_converted_string(const vimconv_T *const sd_conv,
           (tgt) = aetv.vval.v_list; \
         } \
       } while (0)
+// Workaround for silencing linter.
+#define ELSE else
```

Comment:
This workaround is a direct consequence of used macros (more specifically, using `if` as a top-level macros block). I do not like the idea of committing without the fix because I always thought that one of the goals is evenually have zero lint errors in the codebase. There is another variant of fixing: make `CHECK_KEY_IS_STR` start the `if` block. Then `CHECKED_KEY` macros may start with `ELSE`:

```C
#define CHECKED_KEY(entry_name, name, error_desc, tgt, condition, attr, proc) \
  ELSE if (CHECK_KEY(unpacked.data.via.map.ptr[i].key, name)) { \
```

. Using `else` is not possible because it will trigger the same rule, but `ELSE` will then only be placed in one line and not in many.